File: Exercises/101-109 - Small Problems/easy_2/string_assignment.rb
# Take a look at the following code:
#
# ```
# name = 'Bob'
# save_name = name
# name = 'Alice'
# puts name, save_name
# ```
#
# What does this code print out? Think about it for a moment before continuing.
#
# ```
# Alice
# Bob
# ```
#
# you are 100% correct, and the answer should come as no surprise. Now, let's look at something a bit different:
#
# ```
# name = 'Bob'
# save_name = name
# name.upcase!
# puts name, save_name
# ```
#
# What does this print out? Can you explain these results?



##########
# ANSWER #
##########

# name = Bob and save_name points to the same area. Since save_name is not reasssigned to something else,
# it will also be capitalize by the destructive upcase! on name.

# Output: 
# BOB
# BOB

File: Exercises/JavaScript Language Fundamentals/javascript-basics/string-assignment.js
/*
Take a look at the following code:

var name = 'Bob';
var saveName = name;
name = 'Alice';
console.log(name, saveName);
What does this code log to the console? Think about it for a moment before continuing.

If you said that code logged:

Alice Bob
you are 100% correct, and the answer should come as no surprise. Now, let's look at something a bit different:

var name = 'Bob';
var saveName = name;
name.toUpperCase();
console.log(name, saveName);
What does this log? Can you explain these results?
*/

// LOGS
// Bob Bob

// name.toUpperCase() doesn't save its return value to either of the strings and since strings are immutable, there is
// no change.

var name = 'Bob';
var saveName = name;
name.toUpperCase();
console.log(name, saveName);